src/reader/kafka.py

```diff
@@ -0,0 +1,67 @@
+from src.reader.reader import Reader
+
+from kafka import KafkaConsumer
+from kafka.structs import OffsetAndMetadata, TopicPartition
+
+class TopicReader(Reader):
+    """
+    TopicReader implements reading data from a Kafka topic into a stream.
+    Parameters:
+        url:
+            URL that points to a Kafka broker.
+        topic:
+            Kafka topic name to subscribe to.
+        resume_earliest:
+            If True, the consumer will start reading from the last committed offset
+            after a restart. If False, the consumer will start reading at the end of
+            the partition log. Default: True
+        auto_commit:
+            If True, the consumer will automatically commit read offsets every interval.
+            If False, the caller is responsible for committing offsets with the
+            commit() method. Default: True
+        group_id:
+            Kafka consumer group id, this is used internally by the consumer as an
+            identifier for automatic offset committing. Default: 'orca'
+        auto_commit_interval:
+            Time in milliseconds between automatic offset commits. Default: 5000
+    """
```

Comment:
This pushes some design choices onto the caller. In some systems the consumer might just want to keep reading the latest messages and so would set `auto_commit` to stay in sync with the producer(s). In other systems the consumer might want to have the ability to roll back for better error handling and so would set `auto_commit` to false.